File: vetor/exercicio9.c
#include <stdio.h>

int main() {
    int vetor1[30], vetor2[30], vetorResultado[60];
    int i = 0, j = 0, k = 0;

    printf("Digite 30 números inteiros ORDENADOS para o primeiro vetor:\n");
    for (i = 0; i < 30; i++) {
        printf("vetor1[%d]: ", i);
        scanf("%d", &vetor1[i]);
    }

    printf("\nDigite 30 números inteiros ORDENADOS para o segundo vetor:\n");
    for (i = 0; i < 30; i++) {
        printf("vetor2[%d]: ", i);
        scanf("%d", &vetor2[i]);
    }

    i = j = k = 0;
    while (i < 30 && j < 30) {
        if (vetor1[i] < vetor2[j]) {
            vetorResultado[k++] = vetor1[i++];
        } else {
            vetorResultado[k++] = vetor2[j++];
        }
    }

    while (i < 30) {
        vetorResultado[k++] = vetor1[i++];
    }

    while (j < 30) {
        vetorResultado[k++] = vetor2[j++];
    }

    printf("\nVetor intercalado e ordenado:\n");
    for (i = 0; i < 60; i++) {
        printf("%d ", vetorResultado[i]);
    }
    printf("\n");

    return 0;
}

File: vetor/exercicio9.py
conjunto1 = [0] * 30
conjunto2 = [0] * 30
intercalado = []  

print("Digite os 30 números do primeiro conjunto (ordenado):")
for i in range(30):
    conjunto1[i] = int(input(f"Conjunto 1 - Número {i+1}: "))

print("\nDigite os 30 números do segundo conjunto (ordenado):")
for i in range(30):
    conjunto2[i] = int(input(f"Conjunto 2 - Número {i+1}: "))

i = j = 0

while i < 30 and j < 30:
    if conjunto1[i] < conjunto2[j]:
        intercalado.append(conjunto1[i])
        i += 1
    else:
        intercalado.append(conjunto2[j])
        j += 1

while i < 30:
    intercalado.append(conjunto1[i])
    i += 1

while j < 30:
    intercalado.append(conjunto2[j])
    j += 1

print("\nVetor intercalado ordenado:")
for num in intercalado:
    print(num, end=" ")
